### PR TITLE
Allow untyped imports

### LINKED ISSUE #11106: Better handling of untyped node module imports
Consider importing an untyped module, eg. exceljs:

```ts
import exceljs = require('exceljs');
```

This results in an error that the module exceljs is not found. There is at least two ways to fix this:

1. Setting `allowJs: true` and `maxNodeModulesJsDepth: 1`
2. Declare an ambient module in a separate file

I think it would be better if instead we update module resolution semantics such that when we find an appropriate package.json without a typings entry, we store it away. Then if module resolution fails (ie. we also can't find an @types/exceljs), we go back and import the untyped package.json as a JS import. Thoughts?

You might argue that this should only be allowed with --allowJs, but it might also make sense to just always use this behavior when using node resolution rules as you may still not want to allow JS in your own project while still allowing your dependencies to be authored in JS without typings.

### PR DESCRIPTION
**Pull request by andy-ms**
Fixes #11106
Replaces #11446
Made easier by #11704. No longer has to touch `moduleNameResolver.ts`.

### CODE REVIEW

File: src/compiler/diagnosticMessages.json [hunk 1]
```diff
@@ -2869,6 +2869,10 @@
         "category": "Error",
         "code": 6143
     },
+    "A package for '{0}' was found at '{1}', but is untyped. Because '--noImplicitAny' is enabled, this package must have a declaration.": {
```
Comment (mhegazy) [inline src/compiler/diagnosticMessages.json]: how about something more consistent with the rest of the the `--noImplicitAny` error message, e.g.: `Could not find a declaration file for module '{0}'. All imports implicitly have type 'any'.`

File: src/compiler/diagnosticMessages.json [hunk 2]
```diff
@@ -2869,6 +2869,10 @@
         "category": "Error",
         "code": 6143
     },
+    "A package for '{0}' was found at '{1}', but is untyped. Because '--noImplicitAny' is enabled, this package must have a declaration.": {
+        "category": "Error",
+        "code": 6144
```
Comment (mhegazy) [inline src/compiler/diagnosticMessages.json]: make this error code in the `7000` along with the other `--noImplicitAnyErrors`

File: src/compiler/diagnosticMessages.json (hunk 1 above)
Comment (andy-ms) [inline src/compiler/diagnosticMessages.json]: How about: `"Could not find a declaration file for module '{0}'. '{1}' implicitly has an 'any' type."`

Comment (mhegazy) [inline src/compiler/diagnosticMessages.json]: 👍

**Pull request merged**